### PR TITLE
k8sclient: refactor conversion to client-go rest config

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: client/k8sclient/k8sclient.go [hunk 1]
```diff
@@ -62,60 +62,52 @@ func DefaultConfig() Config {
...
-func newRawClientConfig(config Config) *rest.Config {
+func toClientGoRESTConfig(config Config) (*rest.Config, error) {
+	if config.InCluster {
+		config.Logger.Log("debug", "creating in-cluster config")
+		c, err := rest.InClusterConfig()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		return c, nil
+	}
```
Comment (xh3b4sd) [inline client/k8sclient/k8sclient.go]: With this the `MaxQPS` and `MaxBurst` are not configured when the in cluster client is used. What about just copy/paste from the crdclient? I fixed this there.

Comment (kopiczko) [reply client/k8sclient/k8sclient.go]: Done.

File: client/k8sclient/k8sclient.go [hunk 2]
```diff
@@ -62,60 +62,54 @@ func DefaultConfig() Config {
 
 // New returns a Kubernetes Clientset with the provided configuration.
```
Comment (kopiczko) [inline client/k8sclient/k8sclient.go]: This diff is crazy. It's copy/paste anyway.

File: client/crdclient/crdclient.go [hunk 3]
```diff
@@ -60,6 +60,14 @@ func New(config Config) (apiextensionsclient.Interface, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.Address must not be empty when not creating in-cluster client")
 	}
 
+	if config.Address != "" {
+		_, err := url.Parse(config.Address)
+		if err != nil {
+			return nil, microerror.Maskf(invalidConfigError,
+				"config.Address=%s must be a valid URL: %s", config.Address, err)
+		}
+	}
```
Comment (kopiczko) [inline client/crdclient/crdclient.go]: I think this should be invalid config, so I moved it here.

**Pull request merged**